aya-log/src/lib.rs

```diff
@@ -243,9 +243,22 @@ trait Format {
 
 impl Format for &[u8] {
     fn format(&self, last_hint: Option<DisplayHintWrapper>) -> Result<String, ()> {
+        use core::fmt::Write;
         match last_hint.map(|DisplayHintWrapper(dh)| dh) {
-            Some(DisplayHint::LowerHex) => Ok(LowerHexDebugFormatter::format(self)),
-            Some(DisplayHint::UpperHex) => Ok(UpperHexDebugFormatter::format(self)),
+            Some(DisplayHint::LowerHex) => {
```

Comment:
Looks like this change would belong to *HexDebugFormatter and not inline here?

Reply:
I agree. Let's make these changes here:

https://github.com/aya-rs/aya/blob/ab000ad7c3b0715c3cdd9798bd08fc834b114f1a/aya-log/src/lib.rs#L208-L219

https://github.com/aya-rs/aya/blob/ab000ad7c3b0715c3cdd9798bd08fc834b114f1a/aya-log/src/lib.rs#L232-L243

Reply:
I don't think Rust has a formatting option of "print as many zeroes as needed", so it's not really possible to do this in generic constant. And since specialization like in C++ does not exist in (stable) Rust (yet?), I had to hardcode the width of 2 somewhere else.

Reply:
I'm not sure what you mean but UpperHexDebugFormatter (perhaps misnamed) is only used for for `impl Format for &[u8]` so with your change now it becomes dead code. So we might as well do the change there?